app/models/challenge_friend.rb

```diff
@@ -33,8 +34,34 @@ def self.send_challenge(user, params)
       sub_category = SubCategory.find_by(id: params[:sub_category_id])
       return { message: 'Subcategory not found', status: :not_found } if sub_category.nil?
 
-      existing_challenge = ChallengeFriend.find_by(challenger: user, challengee: friend, sub_category: sub_category)
-      return { message: 'You have already sent a challenge to this user with the same subcategory', status: :unprocessable_entity } if existing_challenge
+      existing_pending_challenge = ChallengeFriend.where(
+        "(challenger_id = :user_id AND challengee_id = :friend_id) OR (challenger_id = :friend_id AND challengee_id = :user_id)",
+        {
+          user_id: user.id,
+          friend_id: friend.id
+        }
+      ).where(
+        sub_category: sub_category,
+        challenge_type: 'single_subject',
+        status: 'pending'
+      ).exists?
+
+      return { message: 'A pending challenge with this subcategory already exists between you and the user', status: :unprocessable_entity } if existing_pending_challenge
+    elsif params[:challenge_type] == 'global'
+      existing_pending_global_challenge = ChallengeFriend.where(
+        "(challenger_id = :user_id AND challengee_id = :friend_id) OR (challenger_id = :friend_id AND challengee_id = :user_id)",
+        {
+          user_id: user.id,
+          friend_id: friend.id
+        }
+      ).where(
+        challenge_type: 'global',
+        status: 'pending'
+      ).exists?
+
+      return { message: 'A pending global challenge already exists between you and the user', status: :unprocessable_entity } if existing_pending_global_challenge
+    else
+      return { message: 'Invalid challenge type', status: :unprocessable_entity }
```

Comment:
try to reduce nesting if else conditions. Also change the queries.

Reply:
removed if-else and changed queries.